### DISCUSSION TITLE
Flink MySQL CDC initial模式拉取数据全量数据是走的jdbc吗

### DISCUSSION THREAD

**Discussion by wloverine** [apache/flink-cdc#2445] [Q&A] [answered]
如题，从源码的注释上看，earliest模式是全部读binlog，而对initial模式的描述并没有说明snapshot部分是怎么拉取的，是走的jdbc吗

**Comment by SML0127** [comment, marked as answer]
In initial mode, flink cdc use incremental snapshot. 

An incremental snapshot divides the table into chunks, and fetches the data from the table on a chunk-by-chunk.

During this process, a JDBC connection is created.

A JDBC connection is created before fetching a chunk, and the JDBC connection is released when the task of fetching the chunk is finished.

You can see this in [MySqlSplitReader.java](https://github.com/ververica/flink-cdc-connectors/blob/master/flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSplitReader.java#L227)

Have a nice day😄😄

**Reply by wloverine** [reply]
Thanks for your answer.
